lib/flight_job/commands/edit_script.rb

```diff
@@ -0,0 +1,76 @@
+#==============================================================================
+# Copyright (C) 2021-present Alces Flight Ltd.
+#
+# This file is part of Flight Job.
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which is available at
+# <https://www.eclipse.org/legal/epl-2.0>, or alternative license
+# terms made available by Alces Flight Ltd - please direct inquiries
+# about licensing to licensing@alces-flight.com.
+#
+# Flight Job is distributed in the hope that it will be useful, but
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR
+# IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS
+# OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with Flight Job. If not, see:
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on Flight Job, please visit:
+# https://github.com/openflighthpc/flight-job
+#==============================================================================
+
+require 'tty-editor'
+
+module FlightJob
+  class Commands::EditScript < Command
+    def run
+      # Ensure the script exists up front
+      script
+
+      # Opens vim/emacs/nano commands at the start line, else use the regular editor
+      # NOTE: gedit needs x-forwarding over ssh, but it also supports the +line
+      cmd = if ['vi', 'vim', 'nvim', 'emacs', 'nano', 'gedit'].include?(editor)
+              if start_line
+                "#{editor} +#{start_line}"
```

Comment:
Oh dear! You've got me thinking of how awesome this could be:

For `vim` and `nvim` we could add the following to the command:

```
-c 'let b:scrollofforig=&scrolloff' -c 'set scrolloff=0' -c 'normal! kztj' -c 'let &scrolloff=b:scrollofforig'
```

Without those additions the editor opens as this:


![image](https://user-images.githubusercontent.com/287050/112507015-33da2300-8d86-11eb-9aa5-b08116d01879.png)


With those editions it opens as:

![image](https://user-images.githubusercontent.com/287050/112507131-4e140100-8d86-11eb-9785-6013925e01d8.png)


A little thing, but it really does focus the user's attention on their workload.

Reply:
I'm sure there are options for emacs too, but I don't know them.  If you do, I'd be pleased to have them included, if you don't I'm happy to ship without too.

Reply:
:laughing: I barely supported `emacs` in the first place!